src/pal/src/misc/sysinfo.cpp

```diff
@@ -128,6 +128,52 @@ int GetNumberOfProcessors()
     return nrcpus;
 }
 
+//******************************************************************************
+// Returns the number of processors that a process has been configured to run on
+//******************************************************************************
+int
+PALAPI
+PAL_GetCurrentProcessCpuCount()
+{
+#ifdef HAVE_SCHED_GETAFFINITY
+
+    int pid = getpid();
+    size_t setsize;
+    cpu_set_t* set;
+    int ncpus = CPU_SETSIZE;
+    int rv;
+    do
+    {
+        setsize = CPU_ALLOC_SIZE(ncpus);
```

Comment:
cpu_set_t will work almost always. The kernel may support more cpus than the mask of the clibrary supports. My stock fedora x64 kernel supports up to 8192 while cpu_set_t (glibc) stores up to 1024 cpus. The man page mentions the EINVAL as a result of providing a smaller mask than the kernel. The kernel also takes into account the number of installed cpus, so as long as you don't have more than 1024, it works fine.